### ISSUE TITLE
Move the already existent tools into `bevy_dev_tools`

### ISSUE THREAD

**Issue by pablo-lua** [bevyengine/bevy#12351]
## What problem does this solve or what need does it fill?

After #11341, we now have a canonical place to store our native tools in a better way

## What solution would you like?

The moval of all tools that can be moved to `bevy_dev_tools`, including ones like Aabb gizmos, camera tools, system stepping and so

## What alternative(s) have you considered?

We can move only some tools, and others not.

## Additional context

Message related of which tools we can try to move first:
https://discord.com/channels/691052431525675048/789987800014651412/1215063888064938064

See #11341 for more information.

**Comment by alice-i-cecile** [comment]
Can you update your message with a checklist of tools so we can track them as they get moved? :) These should be moved one tool at a time, to ease burden on reviewers and make sure things don't get stuck pointlessly.

**Comment by pablo-lua** [comment]
I did a little checklist here. Let me know if there is anything that can be improved :)

**Comment by pablo-lua** [comment]
I added Light Gizmos to the checklist: Noticed that this is a debug tool and even adds a plugin to the app

**Comment by mockersf** [comment]
I disagree on moving some gizmos to the dev tools. It should be either all of them or none, and I think I would prefer none

**Comment by pablo-lua** [comment]
> I disagree on moving some gizmos to the dev tools. It should be either all of them or none, and I think I would prefer none.
> 
> We could move the mechanic to draw them automatically for all entities, but the gizmo themself should stay with the others

I agree here that moving something like the ArrowBuilder don't make sense: after all, they are just a "line composer", if I can say that. However, that's not the case of the Aabb gizmos and Light Gizmos: they are plugins Added by the user when they enable bevy_gizmos, they add systems and "event trackers" to entities to drawn above entities automatically, and they even have resources to enable that behavior or disable. So, IMO, those two In specific are exactly the meaning of dev tools.

**Comment by matiqo15** [comment]
I would add:
- [FPS counter](https://bevyengine.org/examples/UI%20(User%20Interface)/text/) - Requires small modification
- [Camera controller](https://github.com/bevyengine/bevy/blob/main/examples/helpers/camera_controller.rs)
- [UI layout debug](https://github.com/bevyengine/bevy/blob/a6ec32aca48a725396c331789b4fff40d8f22da8/crates/bevy_ui/src/layout/debug.rs) - I'm not sure about this one
- #10981
- #8863

**Comment by pablo-lua** [comment]
> Ui Layout debug

I don't think this can be here, because this isn't much of a tool, even if it works for debugging. And It use some fields in types that aren't public. This can be solved, But I don't see much a reason to move it in `bevy_dev_tools` for now.

I think FPS counter can be a controversial decision. But I agree that its a tool and can be registred.

**Comment by alice-i-cecile** [comment]
FPS counter is high on my list :)

I think that the UI layout debug should be refactored so it can fit well into this crate.

**Comment by pablo-lua** [comment]
I changed this issue to track new tools too. If there is anything that I forgot, be free to comment here
 :)

**Comment by cart** [comment]
> I disagree on moving some gizmos to the dev tools. It should be either all of them or none, and I think I would prefer none.

I agree. I think there are better paths:

1. Move gizmo implementations next to their features (ex: bevy_sprite defines its own gizmos)
2. Create a new `bevy_editor_gizmos` crate to be the home for bevy editor gizmos.

"gizmos" are a clear enough feature slice that putting them in the `bevy_dev_tools` grab bag feels wrong.

**Comment by pablo-lua** [comment]
Blocked on the RFC
See https://github.com/bevyengine/rfcs/pull/77

**Comment by NthTensor** [comment]
The dev-tools effort has kind of gone stale, so I am re-organizing all these partial efforts under a new project goal (https://github.com/bevyengine/bevy/issues/23582). Closing this at not planned, but we can revisit as part of that goal.
